pkg/virt-controller/watch/vmi.go

```diff
@@ -477,158 +480,46 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
-			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/conditions", "value": %s }`, string(newConditions)))
+		c.updateVolumeStatus(vmiCopy, pod)
 
-			log.Log.V(3).Object(vmi).Infof("Patching VMI conditions")
-		}
-
-		if !reflect.DeepEqual(vmiCopy.Status.ActivePods, vmi.Status.ActivePods) {
-			newPods, err := json.Marshal(vmiCopy.Status.ActivePods)
-			if err != nil {
-				return err
-			}
-			oldPods, err := json.Marshal(vmi.Status.ActivePods)
-			if err != nil {
-				return err
+		var foundImage string
+		for _, container := range pod.Spec.Containers {
+			if container.Name == "compute" {
+				foundImage = container.Image
+				break
 			}
-
-			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/status/activePods", "value": %s }`, string(oldPods)))
-			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/activePods", "value": %s }`, string(newPods)))
-
-			log.Log.V(3).Object(vmi).Infof("Patching VMI activePods")
 		}
+		vmiCopy = c.setLauncherContainerInfo(vmiCopy, foundImage)
 
-		if vmiPodExists {
-			var foundImage string
-
-			for _, container := range pod.Spec.Containers {
-				if container.Name == "compute" {
-					foundImage = container.Image
-					break
-				}
-			}
-
-			vmiCopy = c.setLauncherContainerInfo(vmiCopy, foundImage)
-
-			if vmiCopy.Status.LauncherContainerImageVersion != vmi.Status.LauncherContainerImageVersion {
-				if vmi.Status.LauncherContainerImageVersion == "" {
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "add", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, vmiCopy.Status.LauncherContainerImageVersion))
-				} else {
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, vmi.Status.LauncherContainerImageVersion))
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, vmiCopy.Status.LauncherContainerImageVersion))
-				}
-			}
-
-			if !reflect.DeepEqual(vmi.Labels, vmiCopy.Labels) {
-				labelBytes, err := json.Marshal(vmiCopy.Labels)
-				if err != nil {
-					return err
-				}
-				origLabelBytes, err := json.Marshal(vmi.Labels)
-				if err != nil {
-					return err
-				}
-
-				if vmi.Labels == nil {
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "add", "path": "/metadata/labels", "value": %s }`, string(labelBytes)))
-				} else {
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/metadata/labels", "value": %s }`, string(origLabelBytes)))
-					patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/metadata/labels", "value": %s }`, string(labelBytes)))
+	case vmi.IsScheduled():
+		// Nothing here
+		break
+	default:
+		return fmt.Errorf("unknown vmi phase %v", vmi.Status.Phase)
+	}
 
-				}
-			}
+	// VMI is owned by virt-controller, so patch instead of update
```

Comment:
I think this should read

`VMI is owned by virt-handler`

Reply:
fixed.